### PR TITLE
[DUOS-512][risk=low] - model fix, better error handling, access review page

### PR DESCRIPTION
**Pull request by rfricke-asymmetrik**
**Addresses**
Partially addresses https://broadinstitute.atlassian.net/browse/DUOS-512 

**Changes**

- Made a small model fix
- Wrapped many chains in exitBlockOnFail to ensure errors properly fail the scenario
- Added more chains to simulate member login and access review

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AccessReviewChains.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.broadinstitute.dsp.consent.services._
+import scala.concurrent.duration._
+
+object AccessReviewChains {
+    val referenceId: String = "darReferenceId"
+    val voteId: String = "voteId"
+
+    def init(additionalHeaders: Map[String, String]): ChainBuilder = {
+        exec(
+            Requests.Votes.getDarVote(200, "${darReferenceId}", "${voteId}", additionalHeaders)
```
Comment (rushtong) [inline automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AccessReviewChains.scala]: We should probably refactor all of the 200's to something like apache's `HttpStatus.SC_OK`. Similarly, look at using the created one for 201, etc.

File: automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AdminChains.scala [hunk 2]
```diff
@@ -26,23 +26,25 @@ object AdminChains {
...
+                val newManageDars: Seq[DataAccessRequestManage] = DarService.setManageRolesByOwner(manageDars)
 
-            val researcherDars: Seq[DataAccessRequestManage] = DarService.getPendingDARsByMostRecent(newManageDars, 2)
-            val electionStatus: ElectionStatus = ElectionStatus(status = "Open", finalAccessVote = false)
-            
-            val newSession = session.set(Requests.Dar.manageDarResponse, researcherDars)
-            newSession.set("electionStatusBody", electionStatus.toJson.compactPrint)
-        }
+                val researcherDars: Seq[DataAccessRequestManage] = DarService.getPendingDARsByMostRecent(newManageDars, 2)
+                val electionStatus: ElectionStatus = ElectionStatus(status = "Open", finalAccessVote = false)
```
Comment (rushtong) [inline automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AdminChains.scala]: Nit: `"Open"` should probably be a constant or an enum somewhere.

File: automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AdminChains.scala [hunk 3]
```diff
@@ -26,23 +26,25 @@ object AdminChains {
         exec(
             Requests.Dar.manageDar(200, additionalHeaders)
         )
-        .exec { session =>
-            implicit val darManageFormat: JsonProtocols.DataAccessRequestManageFormat.type = JsonProtocols.DataAccessRequestManageFormat
-            implicit val userFormat: JsonProtocols.userFormat.type = JsonProtocols.userFormat
-            implicit val dacFormat: JsonProtocols.dacFormat.type = JsonProtocols.dacFormat
-            implicit val electionStatusFormat: JsonProtocols.electionStatusFormat.type = JsonProtocols.electionStatusFormat
+        .exitBlockOnFail {
```
Comment (rushtong) [inline automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AdminChains.scala:30]: Nice

File: automation/src/test/scala/org/broadinstitute/dsp/consent/models/PendingModels.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        totalVotes: Option[Int] = None,
+        votesLogged: Option[Int] = None,
+        rpElectionId: Option[Int] = None,
+        rpVoteId: Option[Int] = None,
+        consentGroupName: Option[String] = None,
+        projectTitle: Option[String] = None,
+        dac: Option[Dac] = None,
+        createDate: Option[Long] = None
+    )
+}
```
Comment (rushtong) [inline automation/src/test/scala/org/broadinstitute/dsp/consent/models/PendingModels.scala:26]: There has been some work on this model class recently with a number of additions and some deprecations. Please add the new items - you'll see it in the v2 endpoint, but also in the schema section of the swagger page.

File: automation/src/test/scala/org/broadinstitute/dsp/consent/scenarios/UserScenarios.scala [hunk 5]
```diff
@@ -3,32 +3,12 @@ package org.broadinstitute.dsp.consent.scenarios
...
-                    NihChains.authenticate(TestConfig.researcherHeader)
-                )
-                .exec(
-                    DarChains.finalDarSubmit(TestConfig.researcherHeader)
-                )
+            scenario("User Login")
+                .exec(Requests.User.me(200, TestConfig.researcherHeader))
         )
     )
 }
```
Comment (rushtong) [inline automation/src/test/scala/org/broadinstitute/dsp/consent/scenarios/UserScenarios.scala:14]: nit: newline

**Review by rushtong — CHANGES_REQUESTED** [review]
Lots of great work, thank you! Very minor feedback inline. Manually tested and works great.

File: automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AccessReviewChains.scala (hunk 1 above)
Comment (rfricke-asymmetrik) [reply automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AccessReviewChains.scala]: Added io.nettify, which is the same framework gatling seems to use. All 200's and 201's are now replaced with OK.code, and CREATED.code

File: automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AdminChains.scala (hunk 2 above)
Comment (rfricke-asymmetrik) [reply automation/src/test/scala/org/broadinstitute/dsp/consent/chains/AdminChains.scala]: Status enum added.

File: automation/src/test/scala/org/broadinstitute/dsp/consent/models/PendingModels.scala (hunk 4 above)
Comment (rfricke-asymmetrik) [reply automation/src/test/scala/org/broadinstitute/dsp/consent/models/PendingModels.scala:26]: What is the v2 endpoint for this? From duos-ui I see a call to dataRequest/cases/pending/{userId}, which looks like it goes to DataRequestCasesResource.java. That api returns PendingCase which has the fields present on this model.

Comment (rushtong) [reply automation/src/test/scala/org/broadinstitute/dsp/consent/models/PendingModels.scala:26]: The V2 endpoint is here: https://consent.dsde-dev.broadinstitute.org/#/Data%20Access%20Request/get_api_dar_manage_v2

However, looking at the swagger now, I am incorrect in what I said above. The new model is not documented accurately. Feel free to ignore this comment - we'll need to address it when we move from V1 to V2 functionality.

Comment (rfricke-asymmetrik) [reply automation/src/test/scala/org/broadinstitute/dsp/consent/models/PendingModels.scala:26]: Okay, cool.

**Review by rushtong — APPROVED** [review]
Thank you, nice work!

**Pull request merged**